src/Compatibility/Core/src/AppHostBuilderExtensions.cs

```diff
@@ -27,10 +58,25 @@ public static class AppHostBuilderExtensions
 			typeof(TimePicker),
 		};
 
-		public static IAppHostBuilder UseFormsCompatibility(this IAppHostBuilder builder, bool registerRenderers = true)
+		static readonly List<(Type Control, Type Renderer)> PendingRenderers = new();
+
+		IMauiHandlersCollection? _handlers;
+
+		public bool UseCompatibilityRenderers { get; set; }
+
+		public void SetHandlersCollection(IMauiHandlersCollection handlersCollection) =>
+			_handlers = handlersCollection;
+
+		public static void AddRenderer(Type control, Type renderer) =>
+			PendingRenderers.Add((control, renderer));
+
+		public void ConfigureServices(HostBuilderContext context, IServiceCollection services)
+		{
+		}
+
+		public void Configure(HostBuilderContext context, IServiceProvider services)
 		{
-			// TODO: This should not be immediately run, but rather a registered delegate with values
-			//       of the Context and LaunchActivatedEventArgs passed in.
+			CompatServiceProvider.SetServiceProvider(services);
```

Comment:
This is the magical line of code that allows the Core service providers to be set so that controls and compat can use it from a static perspective.